packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -367,31 +369,54 @@ export class BlobManager extends TypedEventEmitter<IBlobManagerEvents> {
         );
     }
 
+    /**
+     * Set up a mapping in the redirect table from fromId to toId. Also, notify the runtime that a reference is added
+     * which is required for GC.
+     */
+    private setRedirection(fromId: string, toId: string | undefined) {
+        this.redirectTable.set(fromId, toId);
+        // Notify runtime of a reference added if toId is not undefined. It can be undefined when a blob is uploaded in
+        // detached mode. In this case, the entry will be updated when the blob is updated.
+        if (toId !== undefined) {
+            this.addedBlobReference(this.getBlobGCNodePath(fromId), this.getBlobGCNodePath(toId));
+        }
+    }
+
     private deleteAndEmitsIfEmpty(id: string) {
-        this.pendingBlobs.delete(id);
-        if (!this.hasPendingBlobs) {
-            this.emit("noPendingBlobs");
+        if (this.pendingBlobs.has(id)) {
```

Comment:
Why do we need this if statement here?

Reply:
So that we don't emit "noPendingBlobs" unnecessarily. If there are no pending blobs and `deleteAndEmitsIfEmpty` is called, it would otherwise emit "noPendingBlobs".

Reply:
Not sure about that. Maybe FFX or whiteboard?

Reply:
I'm curious how this change will affect them.

Reply:
I would not be worried. This a corner case which should never happen because entries are only deleted in legit cases. And the behavior with this change is the correct one.

Reply:
Currently no one is listening to noPendingBlobs. WB will eventually do. It's good to check that corner case as well.

Reply:
I'm ok with this being resolved